### PR TITLE
[CELEBORN-914][FOLLOWUP] Add aggressive mode to evict memory shuffle file

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
Add an aggressive mode to evict memory shuffle files.

### Why are the changes needed?
To evict more shuffle files to reduce memory pressure.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
GA.

### CODE REVIEW

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java [hunk 1]
```diff
@@ -232,8 +233,9 @@ private MemoryManager(CelebornConf conf, StorageManager storageManager) {
       memoryFileStorageService.scheduleWithFixedDelay(
           () -> {
             try {
-              if ((memoryFileStorageCounter.sum() >= 0.5 * memoryFileStorageThreshold)
-                  && currentServingState() != ServingState.NONE_PAUSED) {
+              if ((aggressiveMemoryFileEvict && currentServingState() != ServingState.NONE_PAUSED)
```
Comment (RexXiong) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java:236]:
```suggestion
               if (currentServingState() != ServingState.NONE_PAUSED &&(aggressiveMemoryFileEvict
                  || (memoryFileStorageCounter.sum() >= 0.5 * memoryFileStorageThreshold))) 
```

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -3407,6 +3409,17 @@ object CelebornConf extends Logging {
       .checkValue(v => v < Int.MaxValue, "A single memory storage file can not be larger than 2GB")
       .createWithDefaultString("8MB")
 
+  val WORKER_MEMORY_FILE_STORAGE_AGGRESSIVE_EVICT_POLICY: ConfigEntry[Boolean] =
+    buildConf("celeborn.worker.memoryFileStorage.aggressiveEvictPolicy")
```
[the remark this answers is not in the document]
Comment (FMX) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:3413]: Sounds great. There won't be another mode for evicting in memory manager.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 3]
```diff
@@ -3407,6 +3411,25 @@ object CelebornConf extends Logging {
       .checkValue(v => v < Int.MaxValue, "A single memory storage file can not be larger than 2GB")
       .createWithDefaultString("8MB")
 
+  val WORKER_MEMORY_FILE_STORAGE_AGGRESSIVE_EVICT_POLICY_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.worker.memoryFileStorage.aggressiveEvictPolicy.enabled")
```
Comment (RexXiong) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:3415]:
```suggestion
    buildConf("celeborn.worker.memoryFileStorage.evict.aggressiveMode.enabled")
```

**Review by RexXiong — APPROVED** [review]
LGTM, merge to main(v0.6.0) and branch-0.5(v0.5.1)

**Pull request closed without merging**